canonical-web-and-design/maas-ui
# fix(ui): fix KVM list resource usage calculations

## Done

- Fixed KVM list resource usage, taking resource over-commit into account
- Changed the colours for the Overall charts so that Project is blue and Others is green. I think this makes more sense because it's consistent with everywhere else that uses blue to mean the resources used by the VM host itself.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the KVM list and take note of the values that show for Cores and RAM (hover over the chart for actual numbers)
- Check that the numbers match what's found in the KVM details Resources tab in the Overall card.

## Fixes

Fixes #2512 

## Screenshots
### Updated KVM list
![2021-04-22_17-39](https://user-images.githubusercontent.com/25733845/115675217-bfe56780-a391-11eb-8f07-cc95ab068491.png)


### Updated Overall tab
![2021-04-22_17-39_1](https://user-images.githubusercontent.com/25733845/115675240-c2e05800-a391-11eb-8ac5-d4e7ca007919.png)
